openedx/core/djangoapps/external_user_ids/models.py

```diff
@@ -67,26 +70,30 @@ def user_has_external_id(cls, user, type_name):
         return True
 
     @classmethod
-    def add_new_user_id(cls, user, type_name):
+    def add_new_user_id(cls, user, type_name, create_type=False):
         """
         Creates an ExternalId for the User of the type_name provided
         Arguments:
             user: User to create the ID for
             type_name (str): Name of the type of ExternalId
+            create_type (bool): Whether to create the type if it doesn't exist
         Returns:
             (ExternalId): Returns the external id that was created or retrieved
             (Bool): True if the External ID was created, False if it already existed
         """
         try:
             type_obj = ExternalIdType.objects.get(name=type_name)
         except ExternalIdType.DoesNotExist:
-            LOGGER.info(
-                'External ID Creation failed for user {user}, no external id type of {type}'.format(
-                    user=user.id,
-                    type=type_name
+            if create_type:
+                type_obj = ExternalIdType.objects.get_or_create(name=type_name, description='auto generated')[0]
```

Comment:
It seems cleaner to me to explicitly create the type separately, but even if we take this approach I think we should pass in a useful description (instead of "auto generated").

Reply:
+1, I am concerned that the types are auto-generated as mentioned in another comment. Having the same description means that at some point it will invalidate the value in having a description.  The value of the description was understanding why we were creating a new type of external ID.